sentry/src/main/java/io/sentry/util/MapObjectReader.java

```diff
@@ -0,0 +1,350 @@
+package io.sentry.util;
+
+import io.sentry.ILogger;
+import io.sentry.JsonDeserializer;
+import io.sentry.ObjectReader;
+import io.sentry.SentryLevel;
+import io.sentry.vendor.gson.stream.JsonToken;
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.ArrayDeque;
+import java.util.Date;
+import java.util.Deque;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@SuppressWarnings("unchecked")
+public final class MapObjectReader implements ObjectReader {
```

Comment:
there are quite a few local variables to add the `final` keyword to

Reply:
added in https://github.com/getsentry/sentry-java/pull/3215/commits/86baf7fc968a8ccc7dc7e8bdce22bd37e60bf8a6